src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.tsx

```diff
@@ -42,33 +45,42 @@ const GenerateOperatorKeys = () => {
     const isRegisterButtonEnabled = contractOperator.addingNewOperator
         || !inputsData.name
         || !inputsData.publicKey
+        || !inputsData.address
         || displayNameError.shouldDisplay
-        || publicKeyError.shouldDisplay;
+        || publicKeyError.shouldDisplay
+        || addressError.shouldDisplay;
     setRegisterButtonEnabled(!isRegisterButtonEnabled);
     return () => {
       setRegisterButtonEnabled(false);
     };
-  }, [inputsData, displayNameError.shouldDisplay, publicKeyError.shouldDisplay, inputsData.name, inputsData.publicKey]);
+  }, [inputsData, walletStore.accountAddress, displayNameError.shouldDisplay, addressError.shouldDisplay, publicKeyError.shouldDisplay, inputsData.name, inputsData.publicKey]);
 
   const onInputChange = (name: string, value: string) => {
     setInputsData({ ...inputsData, [name]: value });
   };
 
   const onRegisterClick = async () => {
-    const operatorKeys: INewOperatorTransaction = {
-      pubKey: inputsData.publicKey,
-      name: inputsData.name,
-    };
-    contractOperator.setOperatorKeys(operatorKeys);
-    await contractOperator.checkIfOperatorExists(inputsData.publicKey).then((isExists: boolean) => {
-      setOperatorExist(isExists);
-      if (!isExists) {
-        contractOperator.addNewOperator(true).then(() => {
-          applicationStore.setIsLoading(false);
-          history.push(config.routes.OPERATOR.CONFIRMATION_PAGE);
-        });
-      }
-    });
+    console.log('here mother fucker!!!!!');
+    console.log(walletStore.web3.utils.toChecksumAddress(inputsData.address));
+   if (!walletStore.web3.utils.toChecksumAddress(inputsData.address)) {
+     setAddressError({ shouldDisplay: true, errorMessage: 'bla bla bla bs' });
+    } else {
+     const operatorKeys: INewOperatorTransaction = {
+       pubKey: inputsData.publicKey,
+       name: inputsData.name,
+       address: inputsData.address,
+     };
+     contractOperator.setOperatorKeys(operatorKeys);
```

Comment:
Block with variables and constants definitions should be separated from other blocks with function calls and logic with new line.